[
  {
    "id" : "3925d3c1-f7e0-4df3-8a01-cb83522df1aa",
    "prId" : 7400,
    "comments" : [
      {
        "id" : "c85004d9-5bf7-4b2e-aa1e-0517e13ecec1",
        "parentId" : null,
        "author" : {
          "login" : "hurryabit",
          "name" : "Martin Huschenbett",
          "avatarUrl" : "https://avatars0.githubusercontent.com/u/11665611?u=7dbd00d5e7ea53919ed31dc4200f602ebca91c8f&v=4"
        },
        "body" : "If we try to fetch with the wrong type and without proper authorization, which error takes precedence? I would argue the type error should take precedence since it's a static problem.",
        "createdAt" : "2020-09-17T15:48:51Z",
        "updatedAt" : "2020-09-17T15:49:12Z",
        "lastEditedBy" : {
          "login" : "hurryabit",
          "name" : "Martin Huschenbett",
          "avatarUrl" : "https://avatars0.githubusercontent.com/u/11665611?u=7dbd00d5e7ea53919ed31dc4200f602ebca91c8f&v=4"
        },
        "tags" : [
        ]
      },
      {
        "id" : "09efaf6e-0b07-41b3-b709-43dc11ad4d00",
        "parentId" : "c85004d9-5bf7-4b2e-aa1e-0517e13ecec1",
        "author" : {
          "login" : "nickchapman-da",
          "name" : null,
          "avatarUrl" : "https://avatars1.githubusercontent.com/u/49153372?v=4"
        },
        "body" : "For the fetch itself, this is indeed what occurs: The contract fetch is performed; the contract type is checked against what is expected; and finally the fetch node to be inserted in the transaction is created and authorized.\r\n\r\nBut in the testcase which we modified, the authorization which failed was for the parent exercise of the fetch, which should be performed before the fetch, I believe.",
        "createdAt" : "2020-09-17T16:36:17Z",
        "updatedAt" : "2020-09-17T16:36:18Z",
        "lastEditedBy" : {
          "login" : "nickchapman-da",
          "name" : null,
          "avatarUrl" : "https://avatars1.githubusercontent.com/u/49153372?v=4"
        },
        "tags" : [
        ]
      },
      {
        "id" : "9c39f0d1-121a-4a88-a947-c767ff2a313b",
        "parentId" : "c85004d9-5bf7-4b2e-aa1e-0517e13ecec1",
        "author" : {
          "login" : "hurryabit",
          "name" : "Martin Huschenbett",
          "avatarUrl" : "https://avatars0.githubusercontent.com/u/11665611?u=7dbd00d5e7ea53919ed31dc4200f602ebca91c8f&v=4"
        },
        "body" : "Yes, the authorization check for the parent exercise should be performed before the `fetch`. Now, I understand what's going on. It all seems to make sense.",
        "createdAt" : "2020-09-17T16:56:46Z",
        "updatedAt" : "2020-09-17T16:56:46Z",
        "lastEditedBy" : {
          "login" : "hurryabit",
          "name" : "Martin Huschenbett",
          "avatarUrl" : "https://avatars0.githubusercontent.com/u/11665611?u=7dbd00d5e7ea53919ed31dc4200f602ebca91c8f&v=4"
        },
        "tags" : [
        ]
      }
    ],
    "commit" : "a360635b966cdfe27bfd3cd6e156308321238660",
    "line" : 6,
    "diffHunk" : "@@ -1,1 +28,32 @@    })\n\n  test(\"WTFetchFails\", \"Fetching of the wrong type fails\", allocate(SingleParty))(implicit ec => {\n    case Participants(Participant(ledger, party)) =>\n      for {"
  }
]